java-checks/src/main/java/org/sonar/java/checks/security/VerifiedServerHostnamesCheck.java

```diff
@@ -179,15 +192,20 @@ private static boolean isNotFalse(ExpressionTree expression) {
 
   private static class MethodBodyApacheVisitor extends BaseTreeVisitor {
     private boolean isSecured = false;
+    private Symbol variable;
 
     private static final MethodMatcher SET_SSL_CHECK_SERVER_ID = MethodMatcher.create()
       .typeDefinition(APACHE_EMAIL)
       .name("setSSLCheckServerIdentity")
       .addParameter("boolean");
 
+    MethodBodyApacheVisitor(Symbol variable) {
+      this.variable = variable;
+    }
+
     @Override
     public void visitMethodInvocation(MethodInvocationTree mit) {
-      if (SET_SSL_CHECK_SERVER_ID.matches(mit) && isNotFalse(mit.arguments().get(0))) {
+      if (isInvocationOnVariable(mit, variable) && SET_SSL_CHECK_SERVER_ID.matches(mit) && isNotFalse(mit.arguments().get(0))) {
```

Comment:
This test condition leads to FP if we are bad at resolving the variable symbol. I would prefer to replace `isInvocationOnVariable` by something like `isInvocationOnTheGivenVariableButReturnTrueIfYourAreUnsureAboutOnWhatThisMethodIsInvoked` :wink: